src/plugins/data/server/search/routes/call_msearch.ts

```diff
@@ -61,28 +64,27 @@ export function getCallMsearch(dependencies: CallMsearchDependencies) {
     // trackTotalHits is not supported by msearch
     const { track_total_hits: _, ...defaultParams } = await getDefaultSearchParams(uiSettings);
 
-    const body = convertRequestBody(params.body, timeout);
-
-    const promise = shimAbortSignal(
-      esClient.asCurrentUser.msearch(
+    try {
+      const promise = esClient.asCurrentUser.msearch(
         {
-          body,
+          body: convertRequestBody(params.body, timeout),
         },
         {
           querystring: defaultParams,
         }
-      ),
-      params.signal
-    );
-    const response = (await promise) as ApiResponse<{ responses: Array<SearchResponse<any>> }>;
+      );
+      const response = await shimAbortSignal(promise, params.signal);
 
-    return {
-      body: {
-        ...response,
+      return {
         body: {
-          responses: response.body.responses?.map((r: SearchResponse<any>) => shimHitsTotal(r)),
+          ...response,
+          body: {
+            responses: response.body.responses?.map((r: SearchResponse<any>) => shimHitsTotal(r)),
```

Comment:
Nit: Can we use `unknown` rather than `any`?

Reply:
I think there's no point changing this line, since you're changing the shmHitsTotal calls anyway :)